### ISSUE TITLE
samples/net/sockets/echo, echo_async : fails to send the TCP packets

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#2406]
**_Reported by gurucharan p:_**

The application is failing to send the TCP packets because of the following issue 
```
  SIDT    staticIdt.o
  GMMU    mmu_tables.o
  GDT     gdt.o
  BIN     zephyr.bin
To exit from QEMU enter: 'CTRL+a, x'
[QEMU] CPU: qemu32
qemu-system-i386: warning: Unknown firmware file in legacy mode: genroms/multiboot.bin

Connection from 192.0.2.2
***** CPU Page Fault (error code 0x00000002)
Supervisor thread wrote address 0x00000000
PDE: 0x025 Present Read-only User
PTE: 0x000 Non-present Read-only Supervisor
Current thread ID = 0x00405bc0
Faulting segment:address = 0x0008:0x0000b265
eax: 0x00405408, ebx: 0x004112d4, ecx: 0x00000000, edx: 0x004112d4
esi: 0x00000000, edi: 0x00000000, ebp: 0x00411288, esp: 0x00411278
eflags: 0x46
Fatal fault in essential thread! Spinning...
Terminate emulator due to fatal kernel error
/home/intel/guru_zephyr/forked_zephyr/v1.9/zephyr/scripts/Makefile.qemu:26: recipe for target 'run' failed
make[2]: *** [run] Error 1
```

The ICMP ,ARP and socket connection works fine but packet transfer fails. 
I have followed the same steps as given in the README.rst file in the same folder.

(Imported from Jira ZEP-2576)

**Comment by nashif** [comment]
**_by gurucharan p:_**

According to my tests 
with commit ID: 7d01c5ecb72091aef1f0b54029ed3bff4a4d5027  is the one from where the sample application is getting failed.

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

Briefly investigated this and gdb tells that the app is crashing in k_poll() which is called by subsys/net/lib/sockets/sockets.c:_k_fifo_wait_non_empty()
Assining this to Paul as he has better understanding of socket internals.

**Comment by nashif** [comment]
**_by Paul Sokolovsky:_**

Back from vacation, will be looking into this.

**Comment by nashif** [comment]
**_by Paul Sokolovsky:_**

https://github.com/zephyrproject-rtos/zephyr/pull/1323

**Comment by nashif** [comment]
**_by Vikash Kumar:_**

_gurucharan p_  : please verify and close, if fix is as per your expectation.

**Comment by nashif** [comment]
**_by gurucharan p:_**

The issue has been fixed in v1.9.0-rc3. so closing the Jira
